projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -686,7 +686,27 @@ function render_wpcom_subscribe_form( $data, $classes, $styles ) {
 		)
 	);
 
-	$post_access_level = get_post_access_level_for_current_post();
+	$post_access_level          = get_post_access_level_for_current_post();
+	$show_login_access_question = '';
+	$show_login_sign_in_link    = '';
+	$show_login_action_text     = '';
+	if ( $data['show_login'] ) {
+		$is_paid_post               = $post_access_level === 'paid_subscribers' && Jetpack_Memberships::has_connected_account();
+		$show_login_access_question = $is_paid_post
+			? __( 'Already a paid subscriber?', 'jetpack' )
+			: __( 'Already a subscriber?', 'jetpack' );
+
+		require_once JETPACK__PLUGIN_DIR . 'extensions/blocks/premium-content/_inc/subscription-service/include.php';
+		$subscription_service = \Automattic\Jetpack\Extensions\Premium_Content\subscription_service();
+
+		$show_login_sign_in_link = add_query_arg( 'redirect_to', rawurlencode( get_post_permalink() ), 'https://wordpress.com/log-in/link' );
+		if ( ! is_user_logged_in() && empty( $subscription_service->get_token_payload() ) ) {
+			$show_login_action_text = __( 'Log in', 'jetpack' );
+		} else {
+			$show_login_sign_in_link = add_query_arg( 'redirect_to', rawurlencode( $show_login_sign_in_link ), 'https://wordpress.com/log-in?action=logout' );
+			$show_login_action_text  = __( 'Switch accounts', 'jetpack' );
```

Comment:
I don't understand the "switch account" use case...

It will happen when the user is loggued OR payload is not empty.... In both those cases shouldn't we just "not" display the login? Or a logout button?

Reply:
the "Switch account" links to logout, it's a matter of copy 😄, @crisbusquets is giving all this a design iteration (p1694681489008179/1694613846.242699-slack-C052XEUUBL4), let's see what she propose!